### ISSUE TITLE
git http.emptyAuth true not working for Git lfs under Windows TFS-git

### ISSUE THREAD

**Issue by suntong** [git-lfs/git-lfs#3967]
**Describe the bug**

Following up with #3927, I'm accessing in-house private git, which used to be MS TFS, but now as git. With such Windows TFS-git, the `git config --global http.emptyAuth true` setting has no effect on `git lfs`.

**To Reproduce**
1. `git config --global http.emptyAuth true`
2. `git pull` -- AOK
3. Change Windows' password
4. `git pull` again. 

**Expected behavior**

 `git pull` again after changing Windows' password will still work. 
However, the current situation is that, after changing Windows' password,  `git pull` works for normal files, but lfs files will not be working. 

```
fatal: Authentication failed for 'http://td...:8080/tfs/.../_git/.../'
```

**System environment**

GitBash under Windows (v11, but I believe it's true for all Window versions).

**Comment by bk2204** [comment]
Hey,

Are you using NTLM or Kerberos authentication for Git (not Git LFS)?  The `http.emptyAuth` option only applies to libcurl because it won't do any authentication unless a username and password are specified, even when using a single sign-on technique like Kerberos or NTLM SSPI.

On the other hand, Git LFS when using NTLM should try SSPI first on Windows, which is the behavior of `http.emptyAuth`.  In other words, this behavior is the default already.

I suspect what's happening here is that you're using Kerberos for Git and something else (maybe username and password-based NTLM) for Git LFS, so the password is cached somewhere.  Can you provide the output of running your command with `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1`, attaching it if necessary?

**Comment by suntong** [comment]
Sorry for responding late, today was crazy for me, and thus I have to get back to your next Monday. 

Let me try to describe my situation as much as I can now, and confirm it with the command log next Monday.

I don't know how the git, & git lfs authentication is configured on the back end, here is my observation. 

- I think I'm using NTLM SSPI, with single sign-on technique. 
- I don't need to provide username and password to git as long as I'm not having any lfs files, for both pulling and pushing.
- Because of that even after changing Windows' password, `git pull` still works, for normal files.
- Whenever `git pull` is pulling lfs file (for the first time), a username / password windows is popped up. I think the username / password is stored in the default built windows credential mechanism, as there is a default windows command I can bring up the password storage: `rundll32.exe keymgr.dll,KRShowKeyMgr`. 
- Because of such username / password is stored, then git lfs will failed next time when Windows' password is changed, causing the `smudge filter lfs failed`error. 

I'll get back the command log to your next Monday.

Thanks again for helping.

**Comment by suntong** [comment]
Sorry for responding late @bk2204, OP updated.

**Comment by bk2204** [comment]
Okay, so what it looks like is that you're using NTLM for authentication via Negotiate.  You can verify this by base64-decoding the first 8 characters of the authentication data, which will produce a string starting with `NTLM`.

I expect that you're using Windows SSPI, which should theoretically work for Git LFS as well.  However, none of the Git LFS maintainers have access to a WIndows domain, nor do we use NTLM, so while we can accept patches for fixes, we can't troubleshoot them ourselves.  You may find logging out and logging back in after changing your password fixes the problem for you.

You're welcome to try a binary built from `master` (the CI system produces artifacts, including unsigned Windows binaries, if you don't want to build it yourself), which contains Kerberos support and may work better for you.  However, since I only have Unix systems that use Kerberos, I can't guarantee that it will work on Windows.

Sorry I can't be of more help here, but we're kind of limited by the environments that we have access to and develop on.

**Comment by suntong** [comment]
Indeed, base64-decoding the first 8 characters of my authentication data, `TlRMTVNT`, does produce a string starting with `NTLM`: `NTLMSS`. 

I have no idea whether I'm using Windows SSPI or not, and I don't even know what's the simplest git lfs operation that will trigger lfs downloading. I.e., I'll leave the issue open, and defer to any one knows Windows, SSPI & LFS better to scratch such itches. 

Thanks again for all your ***swift*** helps, Brain.
